ta/pkcs11/src/persistent_token.c

```diff
@@ -538,6 +539,40 @@ void release_persistent_object_attributes(struct pkcs11_object *obj)
 	obj->attributes = NULL;
 }
 
+enum pkcs11_rc update_persistent_object_attributes(struct pkcs11_object *obj)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_ObjectHandle hdl = TEE_HANDLE_NULL;
+	uint32_t tee_obj_flags = TEE_DATA_FLAG_ACCESS_WRITE;
+	size_t size = 0;
+
+	if (!obj->attributes)
+		assert(!obj->attributes);
```

Comment:
ok, sorry. To catch gcc warning, I think `assert(obj && obj->attributes);` is the right implementation :), thank you gcc!

and maybe move `size = sizeof(struct obj_attrs) + obj->attributes->attrs_size;` to line 566, before it's used.

Reply:
My bad. Sorry. I don't know what i was thinking to add a negation with obj->attributes in assert. I will correct that.